olive/systems/azureml/aml_system.py

```diff
@@ -323,40 +343,23 @@ def _load_model(self, input_model, output_model_path, pipeline_output_path):
-                    "version": input_model.version,
-                    "model_storage_kind": input_model.model_storage_kind,
-                }
-            )
+            model_path = input_model.model_resource_path
         elif model_json["config"]["model_path"] is not None:
-            downloaded_model_path = pipeline_output_path / model_json["config"]["model_path"]
-            if model_json["type"].lower() == "onnxmodel":
-                # onnx model can have external data
-                output_model_path = ONNXModel.resolve_path(output_model_path)
-                if model_json["config"]["model_storage_kind"] == ModelStorageKind.LocalFolder:
-                    # has external data
-                    # copy the .onnx file along with external data files
-                    shutil.copytree(downloaded_model_path.parent, Path(output_model_path).parent, dirs_exist_ok=True)
-                    # rename the .onnx file to the output_model_path, the downloaded model has "model.onnx" name
-                    (Path(output_model_path).parent / downloaded_model_path.name).rename(output_model_path)
-                else:
-                    # no external data
-                    # just copy over the .onnx file
-                    shutil.copy(downloaded_model_path, output_model_path)
+            resource_type = model_json["config"]["model_path"]["type"]
+            if resource_type in [ResourceType.LocalFile, ResourceType.LocalFolder]:
+                # if the model is downloaded from remote, we need to copy it to the output folder
+                # get the downloaded model path
+                download_model_path = pipeline_output_path / model_json["config"]["model_path"]["config"]["path"]
+                # create a resource path object for the downloaded model
+                downloaded_resource_path = ResourcePath.create_resource_path(download_model_path)
+                # save the downloaded model to the output folder
+                output_model_path = downloaded_resource_path.save_to_dir(
+                    Path(output_model_path).parent, Path(output_model_path).name, True
+                )
+                # create a resource path object for the output model
+                model_path = ResourcePath.create_resource_path(output_model_path)
             else:
-                # handle other model types
-                if Path(output_model_path).suffix != Path(downloaded_model_path).suffix:
-                    output_model_path += Path(downloaded_model_path).suffix
-                if downloaded_model_path.is_file():
-                    # model is a file
-                    shutil.copy(downloaded_model_path, output_model_path)
-                else:
-                    # model is a directory
-                    shutil.copytree(downloaded_model_path, output_model_path, dirs_exist_ok=True)
-            model_path = output_model_path
+                model_path = model_json["config"]["model_path"]
```

Comment:
when the branch could be reached? I am supposing the local branch should always be hit.

Reply:
This is the same case as the one described in the previous comment https://github.com/microsoft/Olive/pull/285#discussion_r1203259172 

We don't have any passes currently that generate such models, but the model path could also be a string name. So this case is for completeness.

Reply:
I think the output model should always be local file or local folder. right?

Reply:
Done. asserts that it is local as discussed offline